src/Simplify_Call.cpp

```diff
@@ -649,7 +649,7 @@ Expr Simplify::visit(const Call *op, ExprInfo *bounds) {
                 return Internal::Call::make(op->type,
                                             Call::if_then_else,
                                             {std::move(cond_value), std::move(true_value), std::move(false_value)},
-                                            op->call_type);
```

Comment:
Isn't preserving the existing call type, whatever it is, better than asserting a new one?

It makes for fewer places to update if we ever change the call type again.

Reply:
Well, maybe? I made this change on the assumption that all of the calls to each intrinsic should be the same call type (#6067), and as of this PR, all of the values of call_type that we pass on here *should* be PureIntrinsic anyway. Happy to change it (back) but not sure it matters much.